Bonsai.Shaders/BindTexture.cs

```diff
@@ -133,10 +144,11 @@ public IObservable<TSource> Process<TSource>(IObservable<TSource> source)
         /// </returns>
         public IObservable<Texture> Process(IObservable<Texture> source)
         {
-            return Process(source, input =>
+            return Process(source, (maybeIndex, _, input) =>
             {
-                GL.ActiveTexture(TextureSlot);
-                GL.BindTexture(TextureTarget, input != null ? input.Id : 0);
+                return maybeIndex is int index
+                    ? ((TextureSequence)input).Textures[index]
+                    : input.Id;
```

Comment:
If I am reading this correctly, I think there is another important sense in which this refactor breaks existing behavior. The intended semantics for the operator is the following:

1. If a texture name is specified, that texture is used, regardless of the type of the `source` sequence.
2. If no texture name is specified, but the type of the input is `Texture`, then the elements of `source` are used for binding.

The reason for giving primacy to texture name is to avoid silent changes in behavior when the type of the input source is "accidentally" a `Texture`. As a concrete example, imagine a situation where we need to bind both a dynamic texture and a static texture for a shader with two texture uniforms:

<img alt="image" src="https://github.com/bonsai-rx/bonsai/assets/5315880/b780bf82-87ca-4af3-b922-d695e20fc11c" height="50px"/>

In the first `BindTexture` we would leave `TextureName` empty so we can bind the dynamic texture to `Texture0`, but in the second `BindTexture` we would specify the name of a static texture we want to bind to `Texture1`. In other words, in the first node we care about the value of the input, whereas in the second node we care only about the timing of the event, i.e. we want to bind the second static texture right after we bind the dynamic texture.

In the previous implementation we could have it both ways, which is something we would like to keep at least for a patch release, although I do agree with you there was an inconsistency in the way that both inputs were captured in the closure that would be nice to resolve.

Reply:
You're right, not sure how I overlooked that. Will fix!